matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/TokenChunkEventPersistor.kt

```diff
@@ -67,14 +67,25 @@ internal class TokenChunkEventPersistor @Inject constructor(
                 .awaitTransaction { realm ->
                     Timber.v("Start persisting ${receivedChunk.events.size} events in $roomId towards $direction")
 
+                    /**
+                     * As per spec
+                     * if no further events are available (either because we have reached the start of the timeline,
+                     * or because the user does not have permission to see any more events), the end property is omitted from the response.
+                     * Synapse is not omitting it and EA was relying on it
+                     */
+                    val chunkEnd = if (receivedChunk.events.isEmpty()) {
```

Comment:
Checking events.is_empty is wrong here. There may be events, but if those are the 'last' events in the room, 'end' will not be set.